src/contract_wrappers/modules/sto/capped_sto_wrapper.ts

```diff
@@ -180,14 +186,59 @@ export default class CappedSTOWrapper extends STOWrapper {
   };
 
   public buyTokens = async (params: BuyTokensParams) => {
+    const txPayableData = {
+      ...params.txData,
+      value: params.value,
+      from: params.from,
+    };
+    const allowBeneficialInvestments = await this.allowBeneficialInvestments();
     assert.isETHAddressHex('beneficiary', params.beneficiary);
-    return (await this.contract).buyTokens.sendTransactionAsync(params.beneficiary, params.txData, params.safetyFactor);
+    assert.isAddressNotZero(params.beneficiary);
+    assert.isBigNumber('value', params.value);
```

Comment:
Not necessary to check this

Reply:
why not?

Reply:
Value is a defined as a BigNumber in the interface. We discussed it before, we don't need to re-check types, typescript does it for us.